gridstatus/miso.py

```diff
@@ -679,8 +679,600 @@ def _get_generation_outages_data(
+        url = f"https://docs.misoenergy.org/marketreports/{query_date.strftime('%Y%m%d')}_rt_bc.xls"
+        logger.info(f"Downloading real-time binding constraints data from {url}")
+
+        excel_file = pd.ExcelFile(url)
+        market_date, publish_date = self._get_constraint_header_dates_from_excel(
+            excel_file,
+        )
+        data = pd.read_excel(
+            excel_file,
+            skiprows=3,
+            dtype={
+                "Constraint Description": object,
+            },
+        )
+
+        data["Interval End"] = pd.to_datetime(
+            market_date.strftime("%Y-%m-%d")
+            + " "
+            + data[
+                "Hour of  Occurrence"
+            ],  # NOTE(kladar): sic, there are two spaces between "Hour of" and "Occurrence"
+        ).dt.tz_localize(
+            self.default_timezone,
+        )  # TODO Confirm that these are hour_ending times
+
+        data["Interval Start"] = data["Interval End"] - pd.Timedelta(minutes=5)
+        data = data.rename(
+            columns={
+                "Branch Name ( Branch Type / From CA / To CA )": "Branch Name",
+            },
+        )
+
+        return data[
+            [
+                "Interval Start",
+                "Interval End",
+                "Flowgate NERC ID",
+                "Constraint ID",
+                "Constraint Name",
+                "Branch Name",
+                "Contingency Description",
+                "Preliminary Shadow Price",
+                "Constraint Description",
+                "Override",
+                "Curve Type",
+                "BP1",
+                "PC1",
+                "BP2",
+                "PC2",
+            ]
+        ]
+
+    # NOTE(kladar): Mostly this method is used for efficient backfilling
+    def get_miso_binding_constraints_real_time_yearly_historical(
+        self,
+        year: int,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """Get the real-time binding constraints data from MISO for a given year.
+
+        Args:
+            year (int): Year
+            verbose (bool, optional): Verbosity. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: Historical real-time binding constraints data
+        """
+        url = f"https://docs.misoenergy.org/marketreports/{year}_rt_bc_HIST.csv"
+        logger.info(f"Downloading real-time binding constraints data from {url}")
+
+        data = pd.read_csv(
+            url,
+            skiprows=2,
+            dtype={
+                "Constraint Description": object,
+            },
+        )
+        data = data.iloc[
+            :-2
+        ]  # NOTE(kladar): The last two rows are report descriptions and disclaimers
+        data["Interval End"] = pd.to_datetime(
+            data["Market Date"] + " " + data["Hour of Occurrence"],
+        ).dt.tz_localize(
+            self.default_timezone,
+        )  # TODO Confirm that these are hour_ending times
```

Comment:
@cwaldoch , confirming, is the "Hour of Occurrence" in the realtime 5 minute historical binding constraint interval-ending?